### PR TITLE
get field type from context by path

### PR DESCRIPTION
**Pull request by ilya-korotya**

### CODE REVIEW

File: merklize/merklize.go [hunk 1]
```diff
@@ -156,6 +160,43 @@ func NewFieldPathFromContext(ctxBytes []byte, ctxType, fieldPath string) (Path,
 	return resPath, nil
 }
 
+func TypeFromContext(ctxBytes []byte, path string) (string, error) {
```
[the remark this answers is not in the document]
Comment (ilya-korotya) [reply merklize/merklize.go:164]: Fixed.
But we have a disabled rule in go linter.

File: loaders/http.go [hunk 2]
```diff
@@ -55,7 +55,11 @@ func (l HTTP) Load(ctx context.Context) (schema []byte, extension string, err er
 
 	//nolint:gosec //reason: remove it after fix for 49366 and uncomment
 	//                       following code
-	defer resp.Body.Close()
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			log.Println("failed close body response:", err.Error())
+		}
+	}()
```
Comment (vmidyllic) [inline loaders/http.go:62]: please, provide link to the fix

File: merklize/merklize.go [hunk 3]
```diff
@@ -20,13 +20,17 @@ import (
 	"github.com/piprate/json-gold/ld"
 )
 
-var defaultHasher Hasher = PoseidonHasher{}
-
-// ErrorFieldIsEmpty is returned when field path to resolve is empty
-var ErrorFieldIsEmpty = errors.New("fieldPath is empty")
+var (
+	defaultHasher Hasher = PoseidonHasher{}
+	numRE                = regexp.MustCompile(`^\d+$`)
```
Comment (vmidyllic) [inline merklize/merklize.go:25]: numRE ?

File: merklize/merklize.go [hunk 4]
```diff
@@ -156,6 +160,44 @@ func NewFieldPathFromContext(ctxBytes []byte, ctxType, fieldPath string) (Path,
...
+	}
+
+	ldCtx, err := ld.NewContext(nil, nil).Parse(ctxObj["@context"])
+	if err != nil {
+		return "", err
+	}
+
+	parts := strings.Split(path, ".")
+
+	for _, term := range parts {
```
Comment (vmidyllic) [inline merklize/merklize.go:178]: can't you reuse pathFromContext function

File: merklize/merklize.go (hunk 3 above)
Comment (ilya-korotya) [reply merklize/merklize.go:25]: For checking,  if the string is a number.

File: merklize/merklize.go (hunk 4 above)
Comment (ilya-korotya) [reply merklize/merklize.go:178]: These functions look the same but use different logic.

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**